src/Microsoft.AspNet.Mvc.Abstractions/Filters/FilterItem.cs

```diff
@@ -20,19 +20,10 @@ public FilterItem(FilterDescriptor descriptor)
             Descriptor = descriptor;
         }
 
-        public FilterItem(FilterDescriptor descriptor, IFilterMetadata filter)
-            : this(descriptor)
-        {
-            if (filter == null)
-            {
-                throw new ArgumentNullException(nameof(filter));
-            }
-
-            Filter = filter;
-        }
-
-        public FilterDescriptor Descriptor { get; set; }
+        public FilterDescriptor Descriptor { get; }
 
         public IFilterMetadata Filter { get; set; }
+
+        public bool ShouldCache { get; set; }
```

Comment:
It's been used in IHttpHandler and folks might be aware of what it means. Not super vested in this name.